src/Analysis/Engine/Impl/ModuleAnalysis.cs

```diff
@@ -234,15 +236,32 @@ private class DefinitionWalker : PythonWalkerWithLocation {
                 if (!scope.EnumerateTowardsGlobal.Any()) {
                     variables = _unit.State.BuiltinModule.GetDefinitions(name.Name).SelectMany(ToVariables);
                 } else {
-                    foreach (var defScope in scope.EnumerateTowardsGlobal
-                        .Where(s => s.ContainsVariable(name.Name) && (s == scope || s.VisibleToChildren || IsFirstLineOfFunction(scope, s, location)))) {
-                        var scopeVariables = GetVariablesInScope(name, defScope).Distinct();
+                    foreach (var s in scope.EnumerateTowardsGlobal) {
```

Comment:
Can we extract the body of `if (expr is NameExpression name) {` into separate method (something like `GetVariablesFromNameExpression`)?

Reply:
Sure